src/front/wgsl/mod.rs

```diff
@@ -1199,20 +1201,20 @@ impl Parser {
                 let mut ready = true;
                 loop {
                     match lexer.next() {
```

Comment:
should we just put `.0` at the end here?

Reply:
We need to capture the span in the `other` pattern in these cases, so we couldn't use `lexer.next().0` directly. Alternatively we could store `lexer.next()` temporarily (to use for `other`) and match on the `temp.0` part only

Reply:
How about this - we return the same thing we used to return - the `Token`, but also the `Lexer` knows about the span of the last token it produced. So the `other` branch can always ask the lexer on what the current span is, but other code paths are unaffected.

Reply:
Yeah sure we could do that. I personally prefer to group token and span together because we'll eventually want to use the spans in lots of places (e.g. errors using spans variable names, spans of function parameter names, the start of a scope, etc.) but `lexer.previous_span()` seems less convenient to use. It would also be nice to avoid the extra state in the lexer but it's not a big deal.

Reply:
ok, sounds reasonable. Let's proceed with your approach, just need a rebase